iterator/src/main/java/com/iluwatar/iterator/list/TreasureChest.java

```diff
@@ -51,7 +48,7 @@ public TreasureChest() {
     items.add(new Item(ItemType.WEAPON, "Dagger of poison"));
   }
 
-  ItemIterator iterator(ItemType itemType) {
+  public Iterator iterator(ItemType itemType) {
```

Comment:
return type should not be raw, it should be `Iterator<Item>`. Previously this was not needed because there was a dedicated interface called `ItemIterator` which only worked for `Items`